### ISSUE TITLE
Start GCM from an absolute directory

### ISSUE THREAD

**Issue by cduverne** [roq-messaging/RoQ#110]
At the startup of a VM we gather the GCM.address without problem.

But when launching the GCM and HCM in the startup script, we get an error message about log4j.properties not being found.

I added a cd to the directory of RoQ before executing the startGCM.sh without success.

I'll launch some more tests, but not sure it can be handled that way.
All ideas are welcome :)

**Comment by sskhiri** [comment]
It would be interesting to have to have the exact log of the error. Can you paste it here ?
It is perhaps related to the hack I use in the script for finding the location of the RoQ installation script. As you can see in https://github.com/roq-messaging/RoQ/blob/develop/roq-core/src/main/resources/scripts/pkg-scripts/startGCM.sh I use

```
ROQ_BIN=$(dirname "$(readlink -f "$(type -P $0 || echo $0)")")
ROQ_HOME=$(dirname $ROQ_BIN)
echo "ROQ BIN Home= $ROQ_HOME"
```

For finding the location of the RoQ Home directory in order to give as parameter the log4j file as 

```
-Dlog4j.configuration="file:$ROQ_HOME/config/log4j.properties
```

So few things to check:
1. Is the log4j file is well in the config directorty ?
2. Is the readlink the best method to find the RoQ home ? notice that this why the script launch out of the RoQ bin directory does not work currently.
3. What is shown by the echo ROQ BIN Home ? This will give an idea whether the current script find correctly the RoQ home

**Comment by cduverne** [comment]
Hello, 

Here attached the error message received from the VM : 
![RoQ Error](https://f.cloud.github.com/assets/1458276/126705/0786c05e-6f6e-11e2-978d-21a24b08d579.jpg)

Will see the other points today.

Cheers

**Comment by sskhiri** [comment]
Indeed, it didi not find the RoQ home. Can you please check the 3 points I mentioned in the last post ? and if you have another smarter way to define RoQ home do not hesitate !

**Comment by cduverne** [comment]
1. Is the log4j file is well in the config directorty ?
   CyD -> Yes, it is !
2. Is the readlink the best method to find the RoQ home ? notice that this why the script launch out of the RoQ bin directory does not work currently.
   CyD -> Seems correct, but in the script you retrieve the script path, which is /bla/bla/bla/RoQ/bin/startGCM.sh and the log4j.properties is in /bla/bla/bla/RoQ/config/log4j.properties and the script can't find the properties file on my machine neither. But RoQ is running.
   On the VM it results by a deadlock of the boot sequence, which isn't a feature :) 
3. What is shown by the echo ROQ BIN Home ? This will give an idea whether the current script find correctly the RoQ home
   CyD -> On the VM : RoQ home = . | on my machine : RoQ home = /

I think we thought this the wrong way, we should put the GCM address as an environment variable in the image at boot via the contextualisation script and use the init.d to start the GCM after the primary boot.

What do you think ?

**Comment by sskhiri** [comment]
Seems to be much MORE clearer in that way !

**Comment by cduverne** [comment]
I updated the script and it works !

But, at startup it can't find ../config/GCM.properties , what is the mechanism ?

Here below the output : 

DEBUG [main](ConfigurationUtils.java:449) - ConfigurationUtils.locate(): base is null, name is ../config/GCM.properties
DEBUG [main](DefaultFileSystem.java:267) - Could not locate file ../config/GCM.properties at null: no protocol: ../config/GCM.properties
DEBUG [main](ConfigurationUtils.java:493) - Loading configuration from the path ../config/GCM.properties
DEBUG [main](AbstractFileConfiguration.java:626) - Base path set to file:///var/lib/RoQ/RoQ/roq/bin/../config/GCM.properties
ERROR [main](FileConfigurationReader.java:55) - Error while reading configuration file - skipped but set the default configuration

Then, "var/lib/RoQ/RoQ/roq/bin/../config/GCM.properties" seems stange...

Thanks in advance
Cyrille
